src/azure-cli/azure/cli/command_modules/appservice/_params.py

```diff
@@ -160,6 +160,7 @@ def load_arguments(self, _):
         c.argument('subnet', help="Name or resource ID of the pre-existing subnet to have the webapp join. The --vnet is argument also needed if specifying subnet by name.")
         c.argument('public_network_access', help="Enable or disable public access to the web app", arg_type=get_enum_type(PUBLIC_NETWORK_ACCESS_MODES))
         c.argument('acr_use_identity', action='store_true', help="Enable or disable pull image from acr use managed identity")
+        c.argument('basic_auth', type=str, help='Enable or disable basic auth.')
```

Comment:
Please refer to `public_network_access` parameter and define it as `get_enum_type`